src/pcre2_compile.c

```diff
@@ -11132,6 +11163,19 @@ PCRE2_ASSERT(ptr <= (pattern + patlen));
 *errorptr = errorcode;
 pcre2_code_free(re);
 re = NULL;
+
+#ifdef SUPPORT_WIDE_CHARS
+if (cb.cranges != NULL)
+  {
+  class_ranges* cranges = cb.cranges;
```

Comment:
The part could be moved after `EXIT` label to be on the safe side. There is an `PCRE2_ASSERT(cb.cranges == NULL);` before that though.